iceoryx_posh/test/integrationtests/test_publisher_subscriber_communication.cpp

```diff
@@ -713,4 +743,36 @@ TEST_F(PublisherSubscriberCommunication_test, PublisherUniqueIdMatchesReceivedSa
     }
 }
 
+#ifdef RUN_BIG_PAYLOAD_TESTS
+
+TEST_F(PublisherSubscriberCommunicationWithBigPayload_test, SendingComplexDataType_BigPayloadStruct)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f612a4ef-5f3a-4951-8f2e-bbc28f6b1a66");
+
+    using Type_t = ComplexDataType<BigPayloadStruct>;
+    auto publisher = createPublisher<Type_t>();
+    auto subscriber = createSubscriber<Type_t>();
+
+    ASSERT_FALSE(publisher->loan()
+                     .and_then([](auto& sample) {
+                         for (uint64_t i = 4242; i < 5353; ++i)
+                         {
+                             sample->complexType.bigPayload[i] = static_cast<uint8_t>(i % 256U);
+                         }
+                         sample.publish();
+                     })
+                     .has_error());
+
+    EXPECT_FALSE(subscriber->take()
+                     .and_then([](auto& sample) {
+                         for (uint64_t i = 4242; i < 5353; ++i)
+                         {
+                             EXPECT_THAT(sample->complexType.bigPayload[i], Eq(static_cast<uint8_t>(i % 256U)));
```

Comment:
Please use `ASSERT_THAT` to break out of the for loop after the first failed check.

Reply:
Oh yeah.
Missed that macro.
Makes a lot more sense than having an if with break